content/learning-paths/mobile-graphics-and-gaming/android_halide/intro.md

```diff
@@ -0,0 +1,165 @@
+* Algorithm. Defines what computations are performed—for example, image filters, pixel transformations, or other mathematical operations on image data.
+* Schedule. Specifies how and where these computations are executed, addressing critical details such as parallel execution, memory usage, caching strategies, and hardware-specific optimizations.
+
+This separation allows developers to rapidly experiment and optimize their code for different hardware architectures or performance requirements without altering the core algorithmic logic.
+
+### Functions, Vars, and Pipelines
+Halide introduces several essential concepts to simplify image-processing pipelines:
+* Functions (Func). Represent discrete computational steps or operations applied across pixels in an image. By defining computations declaratively, functions simplify the description of complex image-processing algorithms.
+* Variables (Var). Symbolize spatial coordinates or dimensions of the data (e.g., horizontal position x, vertical position y, and channel c). Vars serve as symbolic indices to define how computations apply across image data.
+* Pipelines. Comprise interconnected Halide functions that collectively form a complete image-processing workflow. Pipelines clearly define data dependencies, facilitating structured and modular image-processing tasks.
+
+### Scheduling Strategies (Parallelism, Vectorization, Tiling)
+Halide offers several powerful scheduling strategies designed for maximum performance:
+* Parallelism. Executes computations concurrently across multiple CPU cores, significantly reducing execution time for large datasets.
+* Vectorization. Enables simultaneous processing of multiple data elements using SIMD (Single Instruction, Multiple Data) instructions available on CPUs and GPUs, greatly enhancing performance.
+* Tiling. Divides computations into smaller blocks (tiles) optimized for cache efficiency, thus improving memory locality and reducing overhead due to memory transfers.
+
+By combining these scheduling techniques, developers can achieve optimal performance tailored specifically to their target hardware architecture.
+
+## System Requirements and Environment Setup
+To start developing with Halide, your system must meet several requirements and dependencies.
+
+### Installation Options
+Halide can be set up using one of two main approaches:
+* Installing pre-built binaries - pre-built binaries are convenient, quick to install, and suitable for beginners or standard platforms (Windows, Linux, macOS).
+* Building from source - building Halide from source offers greater flexibility, allowing optimization for your specific hardware or operating system configuration.
```

Comment:
True, but I think customizing Halide compiler itself is for very advanced experts, not for our target readers.
Building from source is required if prebuilt binaries are unavailable for your environment or you want to use the latest Halide or LLVM under development